kernel/src/hil/flash.rs

```diff
@@ -12,14 +16,14 @@ pub trait Flash {
     /// when operations complete.
     fn set_client(&self, client: &'static Client);
 
-    /// Read data
-    fn read(&self, offset: usize, buf: &'static mut [u8]);
+    /// Read a page of flash.
+    fn read_page(&self, page_number: usize, buf: &'static mut [u8]) -> ReturnCode;
 
-    /// Write data
-    fn write(&self, offset: usize, buf: &'static mut [u8]);
+    /// Write a page of flash.
+    fn write_page(&self, page_number: usize, buf: &'static mut [u8]) -> ReturnCode;
```

Comment:
So do we want something like:

```
trait Flash<B: AsMut<[u8]> {
    fn write_page(&self, page_number: usize, buf: &'static mut B);
}
```

and in the client you'd say something like:

```rust
struct MyFlashCapsule<B: AsRef<[u8], F: Flash<B>> {
    buffer: TakeCell<&'static mut B>,
    flash: &F
    ...
}

struct<B: AsRef<[u8], F: Flash<B>> MyFlashCapsule<B, F> {
    ...
    self.buf.take.map(|buf| {
            buf.as_mut()[12] = 32;
            self.flash.write_page(1, buf);
    });
    ...
}
```

Which wouldn't be guarnateed to have the right size